### DISCUSSION TITLE
Is there an example of how to use modbus-tcp-slave?

### DISCUSSION THREAD

**Discussion by 129duckflew** [digitalpetri/modbus#80] [General]
In the current latest version 2.0.0-rc3 in the readme file I only see the usage of modbus-master

**Comment by kevinherron** [comment]
There is a server that gets created in the tests but I’ll see about adding a little more explanation to the readme file as well. 

https://github.com/digitalpetri/modbus/blob/6ba5c8270e325232be316296fbdfe7b15c143bef/modbus-tests/src/test/java/modbus/ModbusTcpClientServerIT.java#L48

**Comment by kevinherron** [comment]
There is also a real/production example of it being used in https://github.com/kevinherron/modbus-server-driver

**Comment by 129duckflew** [comment]
The Maven repository jar doesn't seem to contain the NettyTcpServerTransport class, and I introduced the dependencies with the following coordinates
```xml
  <dependency>
            <groupId>com.digitalpetri.modbus</groupId>
            <artifactId>modbus</artifactId>
            <version>2.0.0-RC3</version>
        </dependency>
```
I look at the full files under the introduced dependency com.digitalpetri.modbus.server package as follows
![image](https://github.com/user-attachments/assets/660a820c-9045-49a1-be1e-a3405af1b92f)
Is there a bug in the way I introduced the jar package, thanks for any response

**Comment by kevinherron** [comment]
You need the `modbus-tcp` dependency, which transitively includes `modbus`:

```
<dependency>
    <groupId>com.digitalpetri.modbus</groupId>
    <artifactId>modbus-tcp</artifactId>
    <version>2.0.0-RC3</version>
</dependency>
```
